src/modm/platform/core/cortex/module.lb

```diff
@@ -369,5 +369,4 @@ def build(env):
         env.collect(":build:archflags", "-mfloat-abi={}".format(env["float-abi"]), fpu_spec)
         single_precision = ("-sp-" in fpu_spec)
     if single_precision:
-        env.collect(":build:ccflags", "-fsingle-precision-constant",
-                                      "-Wdouble-promotion")
+        env.collect(":build:ccflags", "-Wdouble-promotion")
```

Comment:
This might make the calculation even more precise at the cost of using a software math library instead of the FPU, since constants are not forced to be `float` instead of `double`. (The STM32F4 FPU is single precision only).

Reply:
Maybe we should create an option in the `:platform:cortex-m` module with a nice description text so that this is more apparent to the user?

Reply:
Nah, you should write `float(1.0/3.0)` or `1.0f/3.0f` to get a single precision floating point constant, which is consistent with the C promotion rules. This flag was just a mistake.

Reply:
Ok, I see the problem with this 🤦. I'll move it to it's own PR.